cocos/navmesh/CCNavMesh.cpp

```diff
@@ -32,20 +32,22 @@
 
 NS_CC_BEGIN
 
+#pragma pack(push,1)
 struct TileCacheSetHeader
 {
-    int magic;
-    int version;
-    int numTiles;
+    int32_t magic;
+    int32_t version;
+    int32_t numTiles;
```

Comment:
all binary file formats should be double-checked. TMX is a text format (xml), so there is no need to check it.
Could you run the Spine binary tests? They work for me on iOS and Mac, but there is one test that doesn't work Ok on Android. Could you test that one? I just want to confirm that it is not my toolchain that is generating errors on android.